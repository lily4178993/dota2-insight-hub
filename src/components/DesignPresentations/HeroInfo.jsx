import React from 'react';
import PropTypes from 'prop-types';
import './heroInfo.css';

function HeroInfo({ data }) {
  const generatePosterlink = (imgUrl) => {
    const nameParam = imgUrl.split('/').pop().replace('.png?', '');
    return `https://cdn.cloudflare.steamstatic.com/apps/dota2/videos/dota_react/heroes/renders/${nameParam}.png`;
  };
  return (
    <div className="hero-info-container">
      <div className="hero-image">
        <img src={generatePosterlink(data.img)} alt={data.localized_name} />
      </div>
      <div className="hero-info">
        <h2>
          {data.localized_name}
          {' '}
          - Basic Stats
        </h2>
        <div className="hero-details">
          <div className="stats-category-container">
            <div className="stats-category">
              <h3>Attributes</h3>
              <ul>
                <li>
                  <strong>Primary Attribute:</strong>
                  {' '}
                  {data.primary_attr}
                </li>
                <li>
                  <strong>Strength:</strong>
                  {' '}
                  {data.base_str}
                </li>
                <li>
                  <strong>Agility:</strong>
                  {' '}
                  {data.base_agi}
                </li>
                <li>
                  <strong>Intelligence:</strong>
                  {' '}
                  {data.base_int}
                </li>
              </ul>
            </div>
            <div className="stats-category">
              <h3>Health & Mana</h3>
              <ul>
                <li>
                  <strong>Base Health:</strong>
                  {' '}
                  {data.base_health}
                </li>
                <li>
                  <strong>Base Health Regeneration:</strong>
                  {' '}
                  {data.base_health_regen}
                </li>
                <li>
                  <strong>Base Mana:</strong>
                  {' '}
                  {data.base_mana}
                </li>
                <li>
                  <strong>Base Mana Regeneration:</strong>
                  {' '}
                  {data.base_mana_regen}
                </li>
              </ul>
            </div>
          </div>
          <div className="stats-category-container">
            <div className="stats-category">
              <h3>Attack</h3>
              <ul>
                <li>
                  <strong>Attack Type:</strong>
                  {' '}
                  {data.attack_type}
                </li>
                <li>
                  <strong>Base Attack Damage:</strong>
                  {' '}
                  {data.base_attack_min}
                  {' '}
                  -
                  {' '}
                  {data.base_attack_max}
                </li>
                <li>
                  <strong>Strength Gain:</strong>
                  {' '}
                  {data.str_gain}
                </li>
                <li>
                  <strong>Agility Gain:</strong>
                  {' '}
                  {data.agi_gain}
                </li>
                <li>
                  <strong>Intelligence Gain:</strong>
                  {' '}
                  {data.int_gain}
                </li>
              </ul>
            </div>
            <div className="stats-category">
              <h3>Combat</h3>
              <ul>
                <li>
                  <strong>Attack Range:</strong>
                  {' '}
                  {data.attack_range}
                </li>
                <li>
                  <strong>Projectile Speed:</strong>
                  {' '}
                  {data.projectile_speed}
                </li>
                <li>
                  <strong>Attack Rate:</strong>
                  {' '}
                  {data.attack_rate}
                </li>
                <li>
                  <strong>Base Attack Time:</strong>
                  {' '}
                  {data.base_attack_time}
                </li>
                <li>
                  <strong>Attack Point:</strong>
                  {' '}
                  {data.attack_point}
                </li>
              </ul>
            </div>
            <div className="stats-category">
              <h3>Movement & Vision</h3>
              <ul>
                <li>
                  <strong>Move Speed:</strong>
                  {' '}
                  {data.move_speed}
                </li>
                <li>
                  <strong>Day Vision:</strong>
                  {' '}
                  {data.day_vision}
                </li>
                <li>
                  <strong>Night Vision:</strong>
                  {' '}
                  {data.night_vision}
                </li>
              </ul>
            </div>
          </div>
          <div className="stats-category-container">
            <div className="stats-category">
              <h3>Turbo Mode</h3>
              <ul>
                <li>
                  <strong>Turbo Picks:</strong>
                  {' '}
                  {data.turbo_picks}
                  {' '}
                  Picks in Turbo mode this month
                </li>
                <li>
                  <strong>Turbo Wins:</strong>
                  {' '}
                  {data.turbo_wins}
                  {' '}
                  Wins in Turbo mode this month
                </li>
              </ul>
            </div>
            <div className="stats-category">
              <h3>Professional Play</h3>
              <ul>
                <li>
                  <strong>Pro Pick:</strong>
                  {' '}
                  {data.pro_pick}
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>

  );
}
HeroInfo.propTypes = {
  data: PropTypes.shape({
    localized_name: PropTypes.string.isRequired,
    img: PropTypes.string.isRequired,
    primary_attr: PropTypes.string.isRequired,
    attack_type: PropTypes.string.isRequired,
    roles: PropTypes.oneOfType([PropTypes.string, PropTypes.array]).isRequired,
    base_health: PropTypes.number.isRequired,
    base_health_regen: PropTypes.number.isRequired,
    base_mana: PropTypes.number.isRequired,
    base_mana_regen: PropTypes.number.isRequired,
    base_armor: PropTypes.number.isRequired,
    base_mr: PropTypes.number.isRequired,
    base_attack_min: PropTypes.number.isRequired,
    base_attack_max: PropTypes.number.isRequired,
    base_str: PropTypes.number.isRequired,
    base_agi: PropTypes.number.isRequired,
    base_int: PropTypes.number.isRequired,
    str_gain: PropTypes.number.isRequired,
    agi_gain: PropTypes.number.isRequired,
    int_gain: PropTypes.number.isRequired,
    attack_range: PropTypes.number.isRequired,
    projectile_speed: PropTypes.number.isRequired,
    attack_rate: PropTypes.number.isRequired,
    base_attack_time: PropTypes.number.isRequired,
    attack_point: PropTypes.number.isRequired,
    move_speed: PropTypes.number.isRequired,
    day_vision: PropTypes.number.isRequired,
    night_vision: PropTypes.number.isRequired,
    turbo_picks: PropTypes.number.isRequired,
    turbo_wins: PropTypes.number.isRequired,
    pro_pick: PropTypes.number.isRequired,
  }).isRequired,
};

export default HeroInfo;
