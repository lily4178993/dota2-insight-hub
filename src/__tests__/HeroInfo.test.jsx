import React from 'react';
import { render } from '@testing-library/react';
import HeroInfo from '../components/DesignPresentations/HeroInfo';

test('HeroInfo component matches snapshot', () => {
  const data = {
    localized_name: 'Pudge',
    img: 'pudge.png',
    primary_attr: 'str',
    attack_type: 'melee',
    roles: ['Disabler', 'Initiator'],
    base_health: 600,
    base_health_regen: 3.5,
    base_mana: 300,
    base_mana_regen: 1.5,
    base_armor: 2,
    base_mr: 25,
    base_attack_min: 50,
    base_attack_max: 70,
    base_str: 25,
    base_agi: 14,
    base_int: 14,
    str_gain: 3.2,
    agi_gain: 1.5,
    int_gain: 1.5,
    attack_range: 150,
    projectile_speed: 900,
    attack_rate: 1.7,
    base_attack_time: 1.7,
    attack_point: 0.5,
    move_speed: 285,
    day_vision: 1800,
    night_vision: 800,
    turbo_picks: 0,
    turbo_wins: 0,
    pro_pick: 200,
  };

  const { asFragment } = render(<HeroInfo data={data} />);
  expect(asFragment()).toMatchSnapshot();
});
