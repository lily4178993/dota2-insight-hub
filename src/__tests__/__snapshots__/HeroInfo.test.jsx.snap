// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroInfo component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="hero-info-container"
  >
    <div
      class="hero-image"
    >
      <img
        alt="Pudge"
        src="https://cdn.cloudflare.steamstatic.com/apps/dota2/videos/dota_react/heroes/renders/pudge.png.png"
      />
    </div>
    <div
      class="hero-info"
    >
      <h2>
        Pudge - Basic Stats
      </h2>
      <div
        class="hero-details"
      >
        <div
          class="stats-category-container"
        >
          <div
            class="stats-category"
          >
            <h3>
              Attributes
            </h3>
            <ul>
              <li>
                <strong>
                  Primary Attribute:
                </strong>
                 str
              </li>
              <li>
                <strong>
                  Strength:
                </strong>
                 25
              </li>
              <li>
                <strong>
                  Agility:
                </strong>
                 14
              </li>
              <li>
                <strong>
                  Intelligence:
                </strong>
                 14
              </li>
            </ul>
          </div>
          <div
            class="stats-category"
          >
            <h3>
              Health & Mana
            </h3>
            <ul>
              <li>
                <strong>
                  Base Health:
                </strong>
                 600
              </li>
              <li>
                <strong>
                  Base Health Regeneration:
                </strong>
                 3.5
              </li>
              <li>
                <strong>
                  Base Mana:
                </strong>
                 300
              </li>
              <li>
                <strong>
                  Base Mana Regeneration:
                </strong>
                 1.5
              </li>
            </ul>
          </div>
        </div>
        <div
          class="stats-category-container"
        >
          <div
            class="stats-category"
          >
            <h3>
              Attack
            </h3>
            <ul>
              <li>
                <strong>
                  Attack Type:
                </strong>
                 melee
              </li>
              <li>
                <strong>
                  Base Attack Damage:
                </strong>
                 50 - 70
              </li>
              <li>
                <strong>
                  Strength Gain:
                </strong>
                 3.2
              </li>
              <li>
                <strong>
                  Agility Gain:
                </strong>
                 1.5
              </li>
              <li>
                <strong>
                  Intelligence Gain:
                </strong>
                 1.5
              </li>
            </ul>
          </div>
          <div
            class="stats-category"
          >
            <h3>
              Combat
            </h3>
            <ul>
              <li>
                <strong>
                  Attack Range:
                </strong>
                 150
              </li>
              <li>
                <strong>
                  Projectile Speed:
                </strong>
                 900
              </li>
              <li>
                <strong>
                  Attack Rate:
                </strong>
                 1.7
              </li>
              <li>
                <strong>
                  Base Attack Time:
                </strong>
                 1.7
              </li>
              <li>
                <strong>
                  Attack Point:
                </strong>
                 0.5
              </li>
            </ul>
          </div>
          <div
            class="stats-category"
          >
            <h3>
              Movement & Vision
            </h3>
            <ul>
              <li>
                <strong>
                  Move Speed:
                </strong>
                 285
              </li>
              <li>
                <strong>
                  Day Vision:
                </strong>
                 1800
              </li>
              <li>
                <strong>
                  Night Vision:
                </strong>
                 800
              </li>
            </ul>
          </div>
        </div>
        <div
          class="stats-category-container"
        >
          <div
            class="stats-category"
          >
            <h3>
              Turbo Mode
            </h3>
            <ul>
              <li>
                <strong>
                  Turbo Picks:
                </strong>
                 0 Picks in Turbo mode this month
              </li>
              <li>
                <strong>
                  Turbo Wins:
                </strong>
                 0 Wins in Turbo mode this month
              </li>
            </ul>
          </div>
          <div
            class="stats-category"
          >
            <h3>
              Professional Play
            </h3>
            <ul>
              <li>
                <strong>
                  Pro Pick:
                </strong>
                 200
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
